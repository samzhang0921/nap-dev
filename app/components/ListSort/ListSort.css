.sortBy {
  font-size: 11px;
  height: 20px;
  padding: 0;
  margin-top: 2px;
  font-weight: 400;
  text-align: center;
  line-height: 20px;
}
