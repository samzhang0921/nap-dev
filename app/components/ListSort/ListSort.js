import React from 'react';
import {render} from 'react-dom';
import style from './ListSort.css';

export default class ListSort extends React.Component {
    constructor(props) {
    super(props)
  }

render() {
  return (
    <div className={style.sortBy}>
                <select name="sortOrder">
                    <option value="0">Sort by</option>
                    <option value="high">Price High to Low</option>
                    <option value="low">Price Low to High</option>
                </select>

        </div>
  );
}
}
