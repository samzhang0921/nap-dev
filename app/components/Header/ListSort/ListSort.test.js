import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import ListSort from './ListSort';

describe('ListSort', () => {

  const updateSort =jest.fn();

  it('should match its snapshot with items', () => {

    const tree = renderer.create(
    <ListSort updateSort={updateSort}/>
  ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change update on the select change', () => {

    const component = mount(<ListSort updateSort={updateSort}/>);
    component.find('select').simulate('change');    
      expect(updateSort).toHaveBeenCalled();
  });
});
