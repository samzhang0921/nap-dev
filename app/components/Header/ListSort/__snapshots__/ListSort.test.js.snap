// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSort should match its snapshot with items 1`] = `
<div
  className="sortBy"
>
  <select
    name="sortOrder"
    onChange={[Function]}
  >
    <option
      value="sort"
    >
      Sort by
    </option>
    <option
      value="low"
    >
      Price Low to High
    </option>
    <option
      value="high"
    >
      Price High to Low
    </option>
  </select>
</div>
`;
